src/Adapter/Driver/Mysqli/Result.php

```diff
@@ -79,17 +81,14 @@ public function initialize($resource, $generatedValue, $isBuffered = null)
 
         if ($isBuffered !== null) {
             $this->isBuffered = $isBuffered;
-        } else {
-            if (
-                $resource instanceof mysqli || $resource instanceof mysqli_result
-                || $resource instanceof mysqli_stmt && $resource->num_rows !== 0
-            ) {
-                $this->isBuffered = true;
-            }
+        } elseif ($resource instanceof mysqli || $resource instanceof mysqli_result
```

Comment:
Would it be worth splitting the conditions across multiple lines so that they're that much easier to read?